lib/buffer.js

```diff
@@ -911,7 +914,7 @@ function bidirectionalIndexOf(buffer, val, byteOffset, encoding, dir) {
   byteOffset = +byteOffset;
   // If the offset is undefined, "foo", {}, coerces to NaN, search whole buffer.
   if (NumberIsNaN(byteOffset)) {
-    byteOffset = dir ? 0 : buffer.length;
+    byteOffset = dir ? 0 : (buffer.length || buffer.byteLength);
```

Comment:
Was this change intentional ?

Is there any cases when `buffer.length` is falsy(0) but `buffer.byteLength` isn't falsy ?

Reply:
@ZYSzys Yeah, see https://github.com/nodejs/node/pull/32770#discussion_r406890316 – `DataView`s don’t have `.length`.

This should probably just be `.byteLength`, though.

Reply:
@addaleax Thanks for explaining, I missed the comments above.

Reply:
> This should probably just be `.byteLength`, though.

I was not sure about this. For `Buffer` and `Uint8Array` it's the same but for other typed arrays like `UInt16Array` it's different. And I don't want to change too much here as it's mainly to avoid the crash.